my-dsl.md

```diff
@@ -1,38 +1,105 @@
+line of simply being a configuration file which Fowler would likely not count
+as a DSL. The reason I'd argue it is not simply a configuration file is there
+needs to be a logical ordering of steps that are user determined.
 
 # Internal or external?
-_Is the language implemented as an internal or external DSL? 
+_Is the language implemented as an internal or external DSL?
 Justify your answer._
 
+The Dockerfile Command Language is an **external** DSL since it does not make
+use (and cannot be used) in a more generic langauge. The Dockerfile must be
+parsed and turned into another language/code. (Lucky for us all of this parsing
+business is open source!)
 
 # Host language
 _What language(s) was (were) used to implement the DSL?_
 
+Looking through the [source code](https://github.com/docker/docker) of the
+whole Docker infrastructure, the parser and builder are written in Golang (and
+some supporting C) which are used to essentially take the English  text commands
+in the Dockerfile and turn them into the lower-level commands to eventually
+build a final container. Docker commands wrap system commands and logical build
+steps under more general, limited commands.
 
 # Benefits
-_Identify one potential benefit of the DSL: how is a programmer's life or a 
+_Identify one potential benefit of the DSL: how is a programmer's life or a
 company's bottom line made easier by the existence of this language?_
 
+Since the Dockerfile Command Language includes a limited set of build commands,
+it makes the overall configuration and process of the final image easier to
+decipher. A developer is forced to compartmentalize the machines build steps
+into Docker commands.
+
+If you go onto the [Dockerhub](https://hub.docker.com/) you can view other
+peoples Dockerfiles. Reading through them, it is pretty easy to get a general
+sense of the goals of building each image.
 
 # Drawbacks
-_Identify one potential drawback of the DSL: what does a programmer or company 
+_Identify one potential drawback of the DSL: what does a programmer or company
 lose by using this DSL instead of a general-purpose language?_
+
+Dockerfile's do not allow custom commands&mdash;instead people will have a
+simple Docker command call to a complex script making it a bit of a scavenger
+hunt to find what exactly is happening from the top-level. It would be nice if a
+user could add custom commands to encapsulate common actions.
+
+For example, let's say I want to have a part of my build that is functionally
+used to seed data (a common occurrence when building images). Instead of writing
+
+```
+RUN './seed-script'
+```
+
+it would be nice to write
+
+```
+SEED data destination
+```
+
+which has a more human-friendly style to it.
+
+> Although I make the suggestion that Docker adds custom commands, it could
+> have the potential to make the Dockerfile Command Langauge too generic and
+> bash-like which Fowler would likely say is a bad thing for something that
+> claims to be a DSL.
```

Comment:
I think I agree with this last point -- adding the ability to create custom commands sounds a lot like the feature creep we were talking about in class today. If the lack of custom commands is really a problem, though, it does beg the question whether a DSL is the right answer here. Would a Turing-complete Bash-like language get the job done more effectively, without the need for much scavenger-hunting?

I guess it probably comes down to tradeoffs: to get the readability and compartmentalization of Dockerfiles, you have to sacrifice a little bit of power. Hopefully as the semester goes on we'll start to develop an idea of when to use a DSL and when to use a more general tool.